patches/tModLoader/Terraria/ModLoader/Engine/LoggingHooks.cs

```diff
@@ -35,29 +36,36 @@ internal static class LoggingHooks
 			if (Logging.f_fileName == null)
 				return;
 
-			new Hook(typeof(StackTrace).GetConstructor(new[] { typeof(Exception), typeof(bool) }), new hook_StackTrace(HookStackTraceEx));
+			_ = new Hook(typeof(StackTrace).GetConstructor(new[] { typeof(Exception), typeof(bool) }), new hook_StackTrace(HookStackTraceEx));
 		}
 
-		private delegate EventHandler SendRequest(object self, HttpWebRequest request);
-		private delegate EventHandler SendRequestHook(SendRequest orig, object self, HttpWebRequest request);
+		private delegate ValueTask<HttpResponseMessage> orig_SendAsyncCore(object self, HttpRequestMessage request, Uri? proxyUri, bool async, bool doRequestAuth, bool isProxyConnect, CancellationToken cancellationToken);
 
-		private delegate void WebOperation_ctor(object self, HttpWebRequest request, object writeBuffer, bool isNtlmChallenge, CancellationToken cancellationToken);
-		private delegate void WebOperation_ctorHook(WebOperation_ctor orig, object self, HttpWebRequest request, object writeBuffer, bool isNtlmChallenge, CancellationToken cancellationToken);
-
-		private delegate bool SubmitRequest(object self, HttpWebRequest request, bool forcedsubmit);
-		private delegate bool SubmitRequestHook(SubmitRequest orig, object self, HttpWebRequest request, bool forcedsubmit);
+		private delegate ValueTask<HttpResponseMessage> hook_SendAsyncCore(orig_SendAsyncCore orig, object self, HttpRequestMessage request, Uri? proxyUri, bool async, bool doRequestAuth, bool isProxyConnect, CancellationToken cancellationToken);
 
 		/// <summary>
 		/// Attempt to hook the .NET internal methods to log when requests are sent to web addresses.
 		/// Use the right internal methods to capture redirects
 		/// </summary>
 		private static void HookWebRequests() {
 			try {
-				// TODO re-implement for Core?
+				// .NET 6
+				var sendAsyncCoreMethodInfo = typeof(HttpClient).Assembly
+					.GetType("System.Net.Http.HttpConnectionPoolManager")
+					?.GetMethod("SendAsyncCore", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Do all calls go through this? Even non synchronous ones? There's `WebClient`, `HttpClient` and `HttpWebRequest` iirc that people may use

Reply:
`HttpClient` does
`HttpWebRequest` delegates to `HttpClient`
`WebClient` I need to check

Reply:
`WebClient` delegates to `HttpWebRequest`

Reply:
Sync calls are actually async calls with `GetAwaiter().GetResult()`